### PR TITLE
DWPF-901 Add risk assessment rule runner and API

### PR DESCRIPTION
**Pull request by SamDudley**
Co-authored-by: Cameron Lamb <CamLamb@users.noreply.github.com>

### CODE REVIEW

File: core/models.py [hunk 1]
```diff
@@ -979,8 +978,14 @@ class CookiesPolicy(models.TextChoices):
 
     task_data = models.JSONField(default=dict)
 
+    # FIXME: duplicate in core.services.risk_assessment.types
+    class RiskLevel(models.IntegerChoices):
+        LOW = 1
+        MODERATE = 2
+        HIGH = 3
```
Comment (SamDudley) [inline core/models.py:985]: If this lives in the service, then model code is depending on service code which doesn't feel right.

However, if it lives here it's better as an `IntegerChoices` object which doesn't play well with `mypy`.

I could change it back to an `IntEnum` or move it back to the service layer and dance around circular imports.

For the moment I've duplicated the code, which works, and perhaps isn't so bad? But I need to have a think about it.

File: core/services/risk_assessment/api.py [hunk 2]
```diff
@@ -36,16 +39,29 @@ def get_risk_level(irap_request: "IrapRequest") -> RiskLevel:
...
+    for step in steps:
+        for field in step.form_class._meta.fields:
+            fields.add(field)
+
+    rules = []
+    for rule in collect_rules(rules_module):
+        if set(rule.fields) <= fields:
+            rules.append(rule)
+
+    return run_rules(irap_request, rules)
```
Comment (SamDudley) [inline core/services/risk_assessment/api.py:64]: This isn't going to be easy to test...

File: tests/core/services/risk_assessment/test_runner.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+from core.services.risk_assessment.runner import Rule, collect_rules, run_rules
+
+
+def test_scratchpad(db):
+    call_command("loaddata", "test_irap_requests.json")
+    irap_request = IrapRequest.objects.get(pk=2)
+    rules = list(collect_rules(rules_module))
+    suite = run_rules(irap_request, rules)
+    print(suite)
+    print(repr(suite.risk_level))
```
Comment (SamDudley) [inline tests/core/services/risk_assessment/test_runner.py:15]: Remove when ready.

File: core/models.py (hunk 1 above)
Comment (CamLamb) [reply core/models.py:985]: > and perhaps isn't so bad?

I disagree 😆
I'm currently thinking that we should just drop the imports from the `core/services/risk_assessment/__init__.py`

I quite like [this style](https://github.com/octoenergy/conventions/blob/7abee9fdd445b062dcc9c054203dc508aafab165/conventions/application.md#exception-imports) which would result in the usage of this looking like:

```python
from core.services.risk_assessment import (
    api as ra_api,  # not settled on the "as" part yet
    types as ra_types,
)

...

level = models.SmallIntegerField(choices=ra_types.RiskLevel.choices, null=True)

...

try:
    ra_api.get_field_risk_data(irap_request, "field_1")
except ra_api.RiskDataError:
    # Do something else.
    pass

```

**Review by CamLamb — APPROVED** [review]

**Pull request merged**